src/components/RoomHeaderAvatars.tsx

```diff
@@ -1,63 +1,58 @@
-import PropTypes from 'prop-types';
 import React, {memo} from 'react';
 import {View} from 'react-native';
-import _ from 'underscore';
 import useStyleUtils from '@hooks/useStyleUtils';
 import useThemeStyles from '@hooks/useThemeStyles';
 import Navigation from '@libs/Navigation/Navigation';
 import CONST from '@src/CONST';
 import ROUTES from '@src/ROUTES';
 import Avatar from './Avatar';
-import avatarPropTypes from './avatarPropTypes';
 import PressableWithoutFocus from './Pressable/PressableWithoutFocus';
 import Text from './Text';
+import type AvatarProps from './types';
 
-const propTypes = {
-    icons: PropTypes.arrayOf(avatarPropTypes),
-    reportID: PropTypes.string,
+type RoomHeaderAvatarsProps = {
+    icons: AvatarProps[];
+    reportID: string;
 };
 
-const defaultProps = {
-    icons: [],
-    reportID: '',
-};
-
-function RoomHeaderAvatars(props) {
-    const navigateToAvatarPage = (icon) => {
+function RoomHeaderAvatars({icons = [], reportID = ''}: RoomHeaderAvatarsProps) {
+    const navigateToAvatarPage = (icon: AvatarProps) => {
         if (icon.type === CONST.ICON_TYPE_WORKSPACE) {
-            Navigation.navigate(ROUTES.REPORT_AVATAR.getRoute(props.reportID));
+            Navigation.navigate(ROUTES.REPORT_AVATAR.getRoute(reportID));
             return;
         }
-        Navigation.navigate(ROUTES.PROFILE_AVATAR.getRoute(icon.id));
+        if (icon.id !== undefined) {
+            Navigation.navigate(ROUTES.PROFILE_AVATAR.getRoute(icon.id));
+        }
     };
 
     const styles = useThemeStyles();
     const StyleUtils = useStyleUtils();
-    if (!props.icons.length) {
+    if (icons.length) {
         return null;
     }
 
-    if (props.icons.length === 1) {
+    if (icons.length === 1) {
         return (
             <PressableWithoutFocus
                 style={[styles.noOutline]}
-                onPress={() => navigateToAvatarPage(props.icons[0])}
+                onPress={() => navigateToAvatarPage(icons[0])}
                 accessibilityRole={CONST.ACCESSIBILITY_ROLE.IMAGEBUTTON}
-                accessibilityLabel={props.icons[0].name}
+                accessibilityLabel={icons[0].name ?? ''}
```

Comment:
What if we make name required? Would that work?

Reply:
I reused [AvatarProps](https://github.com/Expensify/App/blob/79d6cdafe77879a2aa3bc01d62b5a6a3ac105149/src/components/Avatar.tsx#L17) type. Sometimes optional value for a `name` prop is passed to [Avatar](https://github.com/Expensify/App/blob/79d6cdafe77879a2aa3bc01d62b5a6a3ac105149/src/components/Avatar.tsx#L54) component, so I don't think it is possible to make `name` required.

Reply:
Change the type to [Icon](https://github.com/Expensify/App/blob/15aac1ca8ffbafb63346056265f544ef1e1a7a7d/src/types/onyx/OnyxCommon.ts#L26), but I still think name needs to be optional